evalml/tests/component_tests/test_imputer.py

```diff
@@ -291,3 +291,44 @@ def test_imputer_with_none():
     imputer = Imputer()
     transformed = imputer.fit_transform(X, y)
     assert_frame_equal(transformed, expected, check_dtype=False)
+
+
+@pytest.mark.parametrize("data_type", ['pd', 'ww'])
+def test_imputer_all_bool(data_type):
+    X = pd.DataFrame([True, True, False, True, True], dtype=bool)
+    X_expected_arr = pd.DataFrame([True, True, False, True, True], dtype=bool)
+    y = pd.Series([1, 0, 0, 1, 0])
+    if data_type == 'ww':
+        X = ww.DataTable(X)
+        y = ww.DataColumn(y)
+    imputer = Imputer()
+    imputer.fit(X, y)
+    X_t = imputer.transform(X)
+    assert_frame_equal(X_expected_arr, X_t)
+
+    X = pd.DataFrame([True, np.nan, False, np.nan, True], dtype=object)
+    y = pd.Series([1, 0, 0, 1, 0])
+    X_expected_arr = pd.DataFrame([True, True, False, True, True], dtype=object)
+    if data_type == 'ww':
+        X = ww.DataTable(X)
+        y = ww.DataColumn(y)
+    imputer = Imputer()
+    imputer.fit(X, y)
+    X_t = imputer.transform(X)
+    assert_frame_equal(X_expected_arr, X_t)
+
+    X_multi = pd.DataFrame({
+        "bool with nan": pd.Series([True, np.nan, False, np.nan, False], dtype=object),
+        "bool no nan": pd.Series([False, False, False, False, True], dtype=bool),
+    })
+
+    X_multi_expected_arr = pd.DataFrame({
+        "bool with nan": pd.Series([True, False, False, False, False], dtype=object),
+        "bool no nan": pd.Series([False, False, False, False, True], dtype=bool),
+    })
+    if data_type == 'ww':
+        X_multi = ww.DataTable(X_multi)
+    imputer = Imputer()
+    imputer.fit(X_multi, y)
+    X_multi_t = imputer.transform(X_multi)
+    assert_frame_equal(X_multi_expected_arr, X_multi_t)
```

Comment:
Unit tests discussed with @christopherbunn 
1. All cols are bool-type
`X = pd.DataFrame([True, True, False, True, True], dtype=bool)`
2. Imputation strategies work for bool type
`X = pd.DataFrame([True, np.nan, False, np.nan, True], dtype=object)`
`X = pd.DataFrame([True, np.nan, False, np.nan, False], dtype=object)`
3. Multi-type, with at least one bool typed col
```python
    X_multi_expected_arr = pd.DataFrame({
        "bool with nan": pd.Series([True, False, False, False, False], dtype=object),
        "bool no nan": pd.Series([False, False, False, False, True], dtype=bool),
    })
```